app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
There are several issues with this approach:

1. There is no client-side check to prevent a big upload from being sent in the first place
2. The 10 MB limit should use the application settings config

Is there a way just to use the existing Dropzone code to make this drag & drop? That way you can take advantage of existing code GitLab is using for this.